File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Temperature IR Bricklet 2.0

	// Callback function for object temperature callback (parameter has unit °C/10)
	static void ObjectTemperatureCB(BrickletTemperatureIRV2 sender, short temperature)
	{
		Console.WriteLine("Object Temperature: " + temperature/10.0 + " °C");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletTemperatureIRV2 tir =
		  new BrickletTemperatureIRV2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register object temperature callback to function ObjectTemperatureCB
		tir.ObjectTemperatureCallback += ObjectTemperatureCB;

		// Set period for object temperature callback to 1s (1000ms) without a threshold
		tir.SetObjectTemperatureCallbackConfiguration(1000, false, 'x', 0, 0);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your Temperature IR Bricklet 2.0

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var tir = new Tinkerforge.BrickletTemperatureIRV2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Set period for object temperature callback to 1s (1000ms) without a threshold
        tir.setObjectTemperatureCallbackConfiguration(1000, false, 'x', 0, 0);
    }
);

// Register object temperature callback
tir.on(Tinkerforge.BrickletTemperatureIRV2.CALLBACK_OBJECT_TEMPERATURE,
    // Callback function for object temperature callback (parameter has unit °C/10)
    function (temperature) {
        console.log('Object Temperature: ' + temperature/10.0 + ' °C');
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
